pulsar-broker-common/src/main/java/com/yahoo/pulsar/broker/ServiceConfiguration.java

```diff
@@ -291,12 +301,32 @@ public void setWebServicePortTls(int webServicePortTls) {
         this.webServicePortTls = webServicePortTls;
     }
 
-    public boolean isBindOnLocalhost() {
-        return bindOnLocalhost;
+    public String getBindAddress() {
+        if (this.bindAddress == null) {
+            try {
+                LOG.debug("bindAddress not set, attempting default configuration.");
+                this.setBindAddress(InetAddress.getLocalHost().getHostName());
+            } catch (UnknownHostException ex) {
+                LOG.warn(ex.getMessage(), "Using localhost as bindAddress.");
```

Comment:
We should just re-trow an unchecked exception to have the broker exit if the hostname is invalid. Anyway it will probably not reachable on `localhost`.